pkg/queue/instance.go

```diff
@@ -58,7 +59,15 @@ func (q *queueImpl) Push(item Task) {
 	q.cond.Signal()
 }
 
+// Run start the task handler, which should be in a separate go routine.
+// The queue can be closed via stop channel. However when it is being closed,
+// the left elements in the queue will be processed.
+// Note: The queue can be rerun after closed.
 func (q *queueImpl) Run(stop <-chan struct{}) {
+	// enable rerun the queue
```

Comment:
What if they call run between `<-stop` and `q.cond.L.Lock()`? I think it may still end up being closed?

Reply:
make sense, it may cause race now

Reply:
done